File: notification-service/src/main/java/edu/iuh/exchange/notificationservice/domain/model/Notification.java
package edu.iuh.exchange.notificationservice.domain.model;

import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;

@Document(collection = "notifications")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Notification {

    @Id
    private String id;
    
    private String recipientId; // Người nhận
    private String title;
    private String message;
    private String type; // ORDER, CHAT, SYSTEM, REPORT
    private String targetId; // ID of the related entity (OrderId, ProductId, etc.)
    
    @Builder.Default
    private boolean read = false;
    
    @CreatedDate
    @Builder.Default
    private Instant createdAt = Instant.now();
}

File: packages/notification-service/src/models/Notification.js
import mongoose from 'mongoose';

const notificationSchema = new mongoose.Schema(
  {
    recipientId: { type: String, required: true, index: true },
    title: { type: String, required: true },
    message: { type: String, required: true },
    type: {
      type: String,
      enum: ['ORDER', 'CHAT', 'SYSTEM', 'KARMA', 'REPORT', 'PRODUCT'],
      required: true,
      index: true,
    },
    targetId: { type: String, default: null },
    isRead: { type: Boolean, default: false, index: true },
  },
  { timestamps: true },
);

notificationSchema.index({ recipientId: 1, createdAt: -1 });
notificationSchema.index({ recipientId: 1, type: 1, createdAt: -1 });

export const Notification = mongoose.model('Notification', notificationSchema);
